### DISCUSSION TITLE
mem-cpu-load component

### DISCUSSION THREAD

**Discussion by pbnj** [nvim-lualine/lualine.nvim#701] [Ideas]
Similar to [tmux-mem-cpu-load](https://github.com/thewtex/tmux-mem-cpu-load/), it would be nice if there is a way to show memory and cpu consumption in the statusline.

At first, I tried doing it very crudely:

```lua
local function mem_cpu()
  return vim.call('system', 'tmux-mem-cpu-load')
end

require('lualine').setup({
  sections = {
    lualine_x = { mem_cpu }
  }
})
```

But, performance was terrible.

So, I tried reducing the refresh interval to 3 seconds:

```lua

local function mem_cpu()
	return vim.call('system', 'tmux-mem-cpu-load')
end

require('lualine').setup({
	sections = {
		lualine_x = { mem_cpu }
	}
})

if _G.Statusline_timer == nil then
    _G.Statusline_timer = vim.loop.new_timer()
else
    _G.Statusline_timer:stop()
end
_G.Statusline_timer:start(0, 3000, vim.schedule_wrap(function() vim.api.nvim_command('redrawstatus') end))
```

But, that did not seem to be any better.

I am curious if anyone else has thought about a similar idea or knows of a tip/trick to make this work?

**Comment by shadmansaleh** [comment]
You are still executing the command on every redraw by calling redraw in timer you're actually making it worse in this case. Do something like this

- have a cache variable string the result in local scope
- use a function as component that displays from the cache var
- setup a separate function in timer that updates the cached value.
